f2/src/BusinessPage.js

```diff
@@ -8,31 +8,36 @@ import { BusinessInfoForm } from './forms/BusinessInfoForm'
 import { Layout } from './components/layout'
 import { BackButton } from './components/backbutton'
 import { Stack } from '@chakra-ui/core'
+import { useStateValue } from './utils/state'
 
-export const BusinessPage = () => (
-    <Route
-        render={({ history }) => (
-            <Layout>
-                <TrackPageViews />
-                <Stack spacing={10} shouldWrapChildren>
-                    <BackButton route="/impact">
-                        <Trans id="businessInfoPage.backButton" />
-                    </BackButton>
+export const BusinessPage = () => {
+    const [data, dispatch] = useStateValue()
+    const { doneForms } = data
+    return (
+        < Route
+            render={({ history }) => (
+                <Layout>
+                    <TrackPageViews />
+                    <Stack spacing={10} shouldWrapChildren>
+                        <BackButton route="/impact">
+                            <Trans id="businessInfoPage.backButton" />
+                        </BackButton>
 
-                    <Stack spacing={4} role="heading" aria-level="1">
-                        <H1 as="span">
-                            <Trans id="businessPage.title" />
-                        </H1>
-                    </Stack>
+                        <Stack spacing={4} role="heading" aria-level="1">
+                            <H1 as="span">
+                                <Trans id="businessPage.title" />
+                            </H1>
+                        </Stack>
 
-                    <BusinessInfoForm
-                        onSubmit={(client, data) => {
-                            client.writeData({ data: { business: JSON.stringify(data) } })
-                            history.push('/whatHappened')
-                        }}
-                    />
-                </Stack>
-            </Layout>
-        )}
-    />
-)
+                        <BusinessInfoForm
+                            onSubmit={data => {
+                                dispatch({ type: 'saveFormData', data: { business: data } })
```

Comment:
should be `data: { businessInfo: data }`

Reply:
fixed